docs/openshift_horizontal_scaling.md

```diff
@@ -0,0 +1,105 @@
+# Openshift Horizontal Pod Autoscaling (HPA)
+
+It is possible to configure Apisonator to auto scale when deployed in Openshift.
+This document details the required setup.
+
+## Pre-requisites
+
+- Have a working installation of 3scale in an Openshift cluster. You can use [this
+guide](https://github.com/3scale/3scale-operator/blob/master/doc/template-user-guide.md).
+
+## Openshift HPA
+
+Openshift Horizontal Pod Autoscaling (HPA) allows to scale pods horizontally by
+specifying the minimum and maximum number of pods that should be deployed and
+the target CPU utilization. For more details check the [official Openshift
+docs](https://docs.openshift.com/container-platform/4.1/nodes/pods/nodes-pods-autoscaling.html).
+
+## Async
+
+Apisonator can be configured to use a non-blocking Redis client. With this
+feature enabled, a single process should use a whole CPU when properly tuned.
+This means that we can enable Openshift HPA using the CPU utilization as the
+scaling metric.
+
+To learn more about how the async Redis client works, check this [design
+doc](./async.md).
+
+## Setup
+
+### Redis databases
+
+The standard 3scale deployment uses a single Redis pod and sets both the resque
+DB and the data DB in the same Redis process under different DB indexes. This is
+not supported by the async redis library, so we need to either place both DBs in
+the same Redis process or use two separate ones. To configure this, change the
+`CONFIG_REDIS_PROXY` and `REDIS_QUEUES_URL` envs in the listener and worker
```

Comment:
This variable was introduced a long time ago. We can discuss ENV name changes in a separate PR because they require other changes to ensure backwards compatibility.